oak/server/rust/oak_runtime/src/config.rs

```diff
@@ -16,21 +16,36 @@
 
 //! Functionality covering configuration of a Runtime instance.
 
-use crate::{RuntimeConfiguration, RuntimeProxy};
+use crate::{runtime::RuntimeProxy, Runtime, RuntimeConfiguration};
+use log::{error, info};
 use oak_abi::OakStatus;
+use std::sync::Arc;
 use tonic::transport::Certificate;
 
-/// Configures a [`RuntimeProxy`] from the given [`RuntimeConfiguration`] and begins execution.
+/// Configures a [`Runtime`] from the given [`RuntimeConfiguration`] and begins execution.
 ///
 /// Returns a [`RuntimeProxy`] for an initial implicit Node, and a writeable [`oak_abi::Handle`] to
 /// send messages into the Runtime. Creating a new channel and passing the write [`oak_abi::Handle`]
 /// into the runtime will enable messages to be read back out from the [`RuntimeProxy`].
-pub fn configure_and_run(
-    config: RuntimeConfiguration,
-) -> Result<(RuntimeProxy, oak_abi::Handle), OakStatus> {
+pub fn configure_and_run(mut config: RuntimeConfiguration) -> Result<Arc<Runtime>, OakStatus> {
     let proxy = RuntimeProxy::create_runtime(&config.app_config, &config.grpc_config);
+    let config_map = config.config_map.take();
```

Comment:
This seems a premature optimization, I would prefer if we could keep the argument non-mut, even if it means that we cannot "take" this value out of it. In practice I expect the compiler will likely optimize the copy out anyways (but even if it does not, I don't think it's worth compromising immutability). Unless there is some other reason I missed for the mutability.

Reply:
Added a `.clone()`. (I was under the impression that we were potentially going to feed in some quite large things via the config map, so I wanted to be explicitly zero-copy.)

Reply:
Should we consider passing a reference instead perhaps?